src/session.c

```diff
@@ -336,13 +346,17 @@ static CURL *session_get_temp_handle(void) {
 
     log_print(LOG_INFO, SECTION_SESSION_DEFAULT, "Opening temporary cURL session.");
     session = curl_easy_init();
+    update_session_count(true);
```

Comment:
What if session == NULL? I don't think we'll ever end up decrementing `session_count`.